File: Day06/app/src/main/java/com/example/day06/MainActivity.java
package com.example.day06;

import android.os.Bundle;
import android.util.Log;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        Log.d("mytag", "onCreate");

        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });


        findViewById(R.id.btnHello).setOnClickListener(v -> {
            Toast.makeText(this, "Hello", Toast.LENGTH_SHORT).show();
            Log.d("mytag", "Hello");
            TextView tvhello = findViewById(R.id.tvHello);
            tvhello.setText("Hello");
        });

        findViewById(R.id.btnSend).setOnClickListener(view -> {
            TextView tvhello = findViewById(R.id.tvHello);
            EditText etMsg = findViewById(R.id.etMsg);
            tvhello.setText(etMsg.getText());
        });
    }
}

File: Exam21/app/src/main/java/com/example/exam21/MainActivity.kt
package com.example.exam21

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    private lateinit var mbtnSend: Button
    private lateinit var medMsg: EditText
    private lateinit var mtvMsg: TextView

    private lateinit var mSubActivityLaucher: ActivityResultLauncher<Intent>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        mbtnSend = findViewById(R.id.btnSend)
        medMsg = findViewById(R.id.edMsg)
        mtvMsg = findViewById(R.id.tvMsg)

        mSubActivityLaucher = registerForActivityResult(
            ActivityResultContracts.StartActivityForResult()
        ) {
            result ->
            if (result.resultCode == RESULT_OK) {
                val msg = result.data?.getStringExtra("msg")


                mtvMsg.text = "${medMsg.text.toString()} ${msg}"

//                medMsg.setText(medMsg.text.toString() + msg)
//                Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()

            }
        }


        mbtnSend.setOnClickListener {
            val msg = medMsg.text.toString()
            val _intent = Intent(this, SubActivity::class.java)
            _intent.putExtra("msg", msg)
            mSubActivityLaucher.launch(_intent)
        }

    }
}